hp_list/main.c

```diff
@@ -14,6 +14,93 @@
 #include <string.h>
 #include <threads.h>
 
+#ifdef ANALYSIS_OPS
+/*
+ * rtry - the number of retries in the __list_find function.
+ * cons - the number of wait-free contains in the __list_find function that curr
+ * pointer pointed. trav - the number of list element traversal in the
+ * __list_find function. fail - the number of CAS() failures. del  - the number
+ * of list_delete operation failed and restart again. ins  - the number of
+ * list_insert operation failed and restart again.
+ */
+static atomic_uint_fast64_t rtry = 0, cons = 0, trav = 0, fail = 0;
+static atomic_uint_fast64_t del = 0, ins = 0;
+static atomic_uint_fast64_t deletes = 0, inserts = 0;
+
+#define goto_try_again              \
+    do {                            \
+        atomic_fetch_add(&rtry, 1); \
+        goto try_again;             \
+    } while (0)
+#define cons_inc                    \
+    do {                            \
+        atomic_fetch_add(&cons, 1); \
+    } while (0)
+#define trav_inc                    \
+    do {                            \
+        atomic_fetch_add(&trav, 1); \
+    } while (0)
+#define CAS(obj, expected, desired)                             \
+    ({                                                          \
+        atomic_fetch_add(&fail, 1);                             \
+        atomic_compare_exchange_strong(obj, expected, desired); \
+    })
+#define del_inc                    \
+    do {                           \
+        atomic_fetch_add(&del, 1); \
+    } while (0)
+#define ins_inc                    \
+    do {                           \
+        atomic_fetch_add(&ins, 1); \
+    } while (0)
+#define deletes_inc                    \
+    do {                               \
+        atomic_fetch_add(&deletes, 1); \
+    } while (0)
+#define inserts_inc                    \
+    do {                               \
+        atomic_fetch_add(&inserts, 1); \
+    } while (0)
+
+void analysis_func(void)
+{
+    printf("%10s %10s %10s %10s %10s %10s %10s %10s\n", "rtry", "cons", "trav",
+           "fail", "del", "ins", "deletes", "inserts");
+    for (int i = 0; i < 87; i++)
+        printf("-");
+    printf("\n%10ld %10ld %10ld %10ld %10ld %10ld %10ld %10ld\n", rtry, cons,
+           trav, fail, del, ins, deletes, inserts);
+}
+
+#else
+
+#define goto_try_again  \
+    do {                \
+        goto try_again; \
+    } while (0)
+#define cons_inc \
+    do {         \
+    } while (0)
+#define trav_inc \
+    do {         \
+    } while (0)
+#define CAS(obj, expected, desired) \
+    ({ atomic_compare_exchange_strong(obj, expected, desired); })
+#define del_inc \
```

Comment:
Change the name scheme of these macros.
* For atomic operations, such as CAS, use `CAS` and `ATOMIC_INC`.
* For linked list specific operations, such as insertion and deletion, use `TRACE_INS` and `TRACE_DEL`.